jrqs-app/src/java/ua/dp/isd/jrqs/RequirementsService.java

```diff
@@ -27,22 +27,40 @@
 package ua.dp.isd.jrqs;
 
 import java.util.List;
+import ua.dp.isd.jrqs.ESearchBy.SearchBy;
 
 public class RequirementsService {
-    
-    public List<Requirement> getRequirementsByRowId(String RequirementRowIds) {
-        return new RequirementsDAO().getRequirementsByRowIds(RequirementRowIds);
-    }
-    
-    public List<Requirement> getRequirementsByRequirementNumbers(String RequirementNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByRequirementNumbers(RequirementNumbers, LimitBySource);
+
+    private final String rawParamsSearchBy;
+    private final String rawParamsUserInput;
+    private final String rawParamsLimitBySource;
+
+    public RequirementsService(String by, String value, String only) {
+        rawParamsSearchBy = by;
+        rawParamsUserInput = value;
+        rawParamsLimitBySource = only;
     }
-    
-    public List<Requirement> getRequirementsByTmsTaskNumbers(String TmsTaskNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByTmsTaskNumbers(TmsTaskNumbers, LimitBySource);
+
+    public List<Requirement> getRequirements() {
+        if (rawParamsSearchBy == null || rawParamsSearchBy.equals("")) {

```

Comment:
Probably you will often need to check string for null and for “”. There are two good approaches.
First: you stick to using only nulls as default values. In this case you check only for null.
Second: you use both, but have created an util or helper that will check both nullability and emptiness for you. Just to not forget to check one of these two criteria.